R/expand.R

```diff
@@ -113,13 +123,14 @@ crossing <- function(...) {
   }
 
   is_atomic <- map_lgl(x, is_atomic)
+  is_list <- map_lgl(x, is_bare_list)
```

Comment:
I think you could make the code a bit simpler by doing 

```R
is_vector <- map_lgl(x, is_atomic) | map_lgl(x, is_bare_list)
```

Reply:
`ulevels` is being used to handle atomics and it wasn't returning the appropriate results for lists, so I was keeping them separate. Do you have advice on navigating that?

Reply:
I think I'd add a `is.list()` case to `ulevels()` that just returns `unique(x)`